### PR TITLE
WL-3819: Email not sent if user tries to visit non-existant site

### PR DESCRIPTION
**Pull request by nicholaswilson100**
and then uses contact us tool to report a problem with content.

### CODE REVIEW

**Comment by nicholaswilson100** [comment]
Line 186, this ' request.getParameter("siteId")' wasn't doing anything, i.e., equals null.

File: feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 1]
```diff
@@ -175,15 +175,9 @@ else if (!hasViewPermission){
...
-        // when using a URL like /portal/site/{siteId}/page/contact_us this will contain the site ID.
-        String contactUsSiteId = (String) request.getSession().getAttribute("contact.us.origin.site");
-        if (contactUsSiteId!=null) {
-            siteId = contactUsSiteId;
-        }
-        // When inside the !error site the original URL is put as a request parameter and then passed through.
         if (siteId.equals("!error")) { // if site is unavailable then retrieve siteId
-            siteId = request.getParameter("siteId");
+            // When inside the !error site we get the URL of the original site being accesssed
+            siteId = request.getContextPath().split("/portal")[1];
```
Comment (buckett) [inline feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: Array access without check based on something that comes from the request looks like a NullPointerException waiting to happen.

**Review by buckett — CHANGES_REQUESTED** [review]

File: feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java (hunk 1 above)
Comment (nicholaswilson100) [reply feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java]: Fixed

**Comment by nicholaswilson100** [comment]
It turns out it's not really necessary to process the request's contextPath ; we can just pass the whole thing through.

File: feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 2]
```diff
@@ -175,15 +175,9 @@ else if (!hasViewPermission){
...
-        String contactUsSiteId = (String) request.getSession().getAttribute("contact.us.origin.site");
-        if (contactUsSiteId!=null) {
-            siteId = contactUsSiteId;
-        }
-        // When inside the !error site the original URL is put as a request parameter and then passed through.
-        if (siteId.equals("!error")) { // if site is unavailable then retrieve siteId
-            siteId = request.getParameter("siteId");
+        if (siteId.equals("!error")) {
+            // When inside the !error site we get the URL of the original site being accessed
+            return request.getContextPath();
```
Comment (buckett) [inline feedback/src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:180]: This then means that the contact address for the site the user was trying to access doesn't work.

**Pull request merged**